File: typescript/src/find-prime-numbers.ts
const range = function* (end: number) {
  for (let n = 2; n <= end; n++) yield n;
};

const isPrime = (n: number) => {
  for (let d = 2; d * d <= n; d++) {
    if (n % d === 0) return false;
  }
  return n >= 2;
};

const primes = Iterator.from(range(100)).filter(isPrime).toArray();
console.log(primes);

File: rust/src/bin/find_prime_numbers.rs
fn main() {
    let is_prime = |n| {
        if n < 2 {
            return false;
        }

        let mut d = 2;
        while d * d <= n {
            if n % d == 0 {
                return false;
            }
            d += 1;
        }
        true
    };

    let primes = (2..=100).filter(|&n| is_prime(n)).collect::<Vec<_>>();
    println!("{primes:?}");
}
